### PR TITLE
Implement the remaining dt methods

### PR DESCRIPTION
**Pull request by icexelloss**
Implement the remaining dt methods per issue #282

### CODE REVIEW

File: databricks/koalas/tests/test_series_datetime.py [hunk 1]
```diff
@@ -28,7 +28,7 @@ class SeriesDateTimeTest(ReusedSQLTestCase, SQLTestUtils):
 
     @property
     def pdf1(self):
-        date1 = pd.Series(pd.date_range('2012-1-1 12:00:00', periods=3, freq='M'))
+        date1 = pd.Series(pd.date_range('2012-1-1 12:45:31', periods=3, freq='M'))
```
Comment (icexelloss) [inline databricks/koalas/tests/test_series_datetime.py]: For testing some of the rounding functions

File: databricks/koalas/datetimes.py [hunk 2]
```diff
@@ -188,7 +191,111 @@ def dayofyear(self) -> 'ks.Series':
...
+        """
+        The quarter of the date.
+        """
+        return _wrap_accessor_pandas(
+            self, lambda s: s.dt.quarter, LongType()).alias(self._data.name)
+
+    @property
+    def is_month_start(self) -> 'ks.Series':
+        """
+        Indicates whether the date is the first day of the month.
```
Comment (HyukjinKwon) [inline databricks/koalas/datetimes.py]: This seems having a doctest in Pandas side - https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.dt.is_month_start.html#pandas.Series.dt.is_month_start

File: databricks/koalas/datetimes.py [hunk 3]
```diff
@@ -188,7 +191,111 @@ def dayofyear(self) -> 'ks.Series':
...
+        """
+        return _wrap_accessor_pandas(
+            self, lambda s: s.dt.is_month_start, BooleanType()).alias(self._data.name)
+
+    @property
+    def is_month_end(self) -> 'ks.Series':
+        """
+        Indicates whether the date is the last day of the month.
+        """
```
Comment (HyukjinKwon) [inline databricks/koalas/datetimes.py]: this too https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.dt.is_month_end.html#pandas.Series.dt.is_month_end

File: databricks/koalas/datetimes.py [hunk 4]
```diff
@@ -188,7 +191,111 @@ def dayofyear(self) -> 'ks.Series':
...
+        return _wrap_accessor_pandas(
+            self, lambda s: s.dt.is_month_end, BooleanType()).alias(self._data.name)
+
+    @property
+    def is_quarter_start(self) -> 'ks.Series':
+        """
+        Indicator for whether the date is the first day of a quarter.
```
Comment (HyukjinKwon) [inline databricks/koalas/datetimes.py]: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.dt.is_quarter_start.html#pandas.Series.dt.is_quarter_start

File: databricks/koalas/datetimes.py [hunk 5]
```diff
@@ -188,7 +191,111 @@ def dayofyear(self) -> 'ks.Series':
...
+        """
+        return _wrap_accessor_pandas(
+            self, lambda s: s.dt.is_quarter_start, BooleanType()).alias(self._data.name)
+
+    @property
+    def is_quarter_end(self) -> 'ks.Series':
+        """
+        Indicator for whether the date is the last day of a quarter.
+        """
```
Comment (HyukjinKwon) [inline databricks/koalas/datetimes.py]: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.dt.is_quarter_end.html#pandas.Series.dt.is_quarter_end

File: databricks/koalas/datetimes.py [hunk 6]
```diff
@@ -188,7 +191,111 @@ def dayofyear(self) -> 'ks.Series':
...
+        return _wrap_accessor_pandas(
+            self, lambda s: s.dt.is_quarter_end, BooleanType()).alias(self._data.name)
+
+    @property
+    def is_year_start(self) -> 'ks.Series':
+        """
+        Indicate whether the date is the first day of a year.
+        """
+        return _wrap_accessor_pandas(
+            self, lambda s: s.dt.is_year_start, BooleanType()).alias(self._data.name)
```
Comment (HyukjinKwon) [inline databricks/koalas/datetimes.py]: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.dt.is_year_start.html#pandas.Series.dt.is_year_start

File: databricks/koalas/datetimes.py [hunk 7]
```diff
@@ -188,7 +191,111 @@ def dayofyear(self) -> 'ks.Series':
...
+
+    @property
+    def is_year_end(self) -> 'ks.Series':
+        """
+        Indicate whether the date is the last day of the year.
+        """
```
Comment (HyukjinKwon) [inline databricks/koalas/datetimes.py]: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.dt.is_year_end.html#pandas.Series.dt.is_year_end

File: databricks/koalas/datetimes.py [hunk 8]
```diff
@@ -188,7 +191,111 @@ def dayofyear(self) -> 'ks.Series':
...
+        return _wrap_accessor_pandas(
+            self, lambda s: s.dt.is_year_end, BooleanType()).alias(self._data.name)
+
+    @property
+    def is_leap_year(self) -> 'ks.Series':
+        """
+        Boolean indicator if the date belongs to a leap year.
+        """
```
Comment (HyukjinKwon) [inline databricks/koalas/datetimes.py]: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.dt.is_leap_year.html#pandas.Series.dt.is_leap_year

File: databricks/koalas/datetimes.py [hunk 9]
```diff
@@ -188,7 +191,111 @@ def dayofyear(self) -> 'ks.Series':
...
+        Convert tz-aware Datetime column from one time zone to another.
+        """
+        # tz-aware datetime doesn't exist in Spark
+        raise NotImplementedError()
+
+    def normalize(self) -> 'ks.Series':
+        """
+        Convert times to midnight.
+        """
```
Comment (HyukjinKwon) [inline databricks/koalas/datetimes.py]: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.dt.normalize.html#pandas.Series.dt.normalize

File: databricks/koalas/datetimes.py [hunk 10]
```diff
@@ -198,3 +305,53 @@ def strftime(self, date_format) -> 'ks.Series':
             lambda x: x.dt.strftime(date_format),
             StringType()
         ).alias(self.name)
+
+    def round(self, freq) -> 'ks.Series':
+        """
+        Perform round operation on the data to the specified freq.
+        """
```
Comment (HyukjinKwon) [inline databricks/koalas/datetimes.py]: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.dt.round.html#pandas.Series.dt.round

File: databricks/koalas/datetimes.py [hunk 11]
```diff
@@ -198,3 +305,53 @@ def strftime(self, date_format) -> 'ks.Series':
...
+        return _wrap_accessor_pandas(
+            self,
+            lambda x: x.dt.round(freq),
+            TimestampType()
+        ).alias(self.name)
+
+    def floor(self, freq) -> 'ks.Series':
+        """
+        Perform floor operation on the data to the specified freq.
+        """
```
Comment (HyukjinKwon) [inline databricks/koalas/datetimes.py]: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.dt.floor.html#pandas.Series.dt.floor

File: databricks/koalas/datetimes.py [hunk 12]
```diff
@@ -198,3 +305,53 @@ def strftime(self, date_format) -> 'ks.Series':
...
+
+    def ceil(self, freq) -> 'ks.Series':
+        """
+        Perform ceil operation on the data to the specified freq.
+        """
+        return _wrap_accessor_pandas(
+            self,
+            lambda x: x.dt.ceil(freq),
+            TimestampType()
+        ).alias(self.name)
```
Comment (HyukjinKwon) [inline databricks/koalas/datetimes.py]: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.dt.ceil.html#pandas.Series.dt.ceil

File: databricks/koalas/datetimes.py [hunk 13]
```diff
@@ -198,3 +305,53 @@ def strftime(self, date_format) -> 'ks.Series':
...
+
+    def month_name(self, locale=None) -> 'ks.Series':
+        """
+        Return the month names of the DateTimeIndex with specified locale.
+        """
```
Comment (HyukjinKwon) [inline databricks/koalas/datetimes.py]: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.dt.month_name.html#pandas.Series.dt.month_name

File: databricks/koalas/datetimes.py [hunk 14]
```diff
@@ -198,3 +305,53 @@ def strftime(self, date_format) -> 'ks.Series':
...
+        return _wrap_accessor_pandas(
+            self,
+            lambda x: x.dt.month_name(locale),
+            StringType()
+        ).alias(self.name)
+
+    def day_name(self, locale=None) -> 'ks.Series':
+        """
+        Return the day names of the DateTimeIndex with specified locale.
+        """
```
Comment (HyukjinKwon) [inline databricks/koalas/datetimes.py]: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.dt.day_name.html#pandas.Series.dt.day_name

**Comment by icexelloss** [comment]
Thanks! @HyukjinKwon

**Comment by ppakawatk** [comment]
Hi,

Is it possible to add/ subtract days in Koalas DataFrame (series)? 
I got these errors using timedelta/ relativedelta:
'datetime.timedelta' object has no attribute '_get_object_id'
'relativedelta' object has no attribute '_get_object_id'

**Comment by HyukjinKwon** [comment]
Please file an issue with the codes you ran.

**Comment by ppakawatk** [comment]
Sorry.
`import databricks.koalas as ks
import datetime as dt
test_df = ks.DataFrame([dt.date(2019, 10, 24)], columns={'date'})
test_df['date'] + dt.timedelta(days=1)`

**Comment by harupy** [comment]
@ppakawatk 
I found a workaround.

```
import databricks.koalas as ks
import datetime

test_df = ks.DataFrame([(datetime.datetime(2019, 10, 24))], columns=['date'])
test_df['date'] = test_df['date'].map(lambda d: d + datetime.timedelta(days=1))
```

**Comment by ppakawatk** [comment]
> @ppakawatk
> I found a workaround.
> 
> ```python
> import databricks.koalas as ks
> import datetime
> 
> test_df = ks.DataFrame([(datetime.datetime(2019, 10, 24))], columns=['date'])
> test_df['date'] = test_df['date'].map(lambda d: d + datetime.timedelta(days=1))
> 
> >>> test_df
>         date
> 0 2019-10-25
> ```

Thank you!, @harupy.

**Pull request merged**